tests/invoice_utils/test_web.py

```diff
@@ -326,3 +325,20 @@ def test_template_related_variables_with_bad_inputs_raise_errors(
 ):
     with pytest.raises(expected_error):
         http.post("/invoice", json=email_invoice_request_body)
+
+
+@pytest.mark.parametrize(
+    "environment",
+    [
+        ({"INVOICE_UTILS_INVOICE_DIR": "bad_directory"}),
+        ({"INVOICE_UTILS_DEFAULT_RULE_TEMPLATE_NAME": "bad_name.json"})
+    ],
+    indirect=["environment"]
+)
+def test_app_does_not_start_with_bad_config(environment, monkeypatch, template_repo, email_invoice_request_body):
+    with patch("dotenv.load_dotenv", MagicMock(name="load_dotenv")):
+        import invoice_utils.web as web
+
+        with pytest.raises(HTTPException):
+            with TestClient(web.app) as client:
```

Comment:
Note: We can only test the startup event using a context manager [docs](https://fastapi.tiangolo.com/advanced/testing-events/), if I did this in the http fixture, all tests would be impacted by the checks at startup.

Reply:
when the rule template specified by `INVOICE_UTILS_RULE_TEMPLATE_NAME` environment variable name does not exist, We should fall back to the `DEFAULT_RULE_TEMPLATE_NAME` constant.
We can achieve this coercion by using the `template_repository()` dependency in a [lifespan function](https://fastapi.tiangolo.com/advanced/events/#lifespan) for the FastAPI app.
This assumes that the actual rule name being used is mutable and changed during app startup accordingly.

Reply:
@koosie0507 the behaviour is the same for the lifespan route, the tests that do not use TestClient in a context manager (so all tests using http) do not actually make use of the lifespan code.
I tested that separately in 2 tests which specifically test that behaviour.